src/smt/sygus_solver.cpp

```diff
@@ -0,0 +1,403 @@
+             << std::endl;
+    Trace("check-synth-sol") << "Expanded assertion " << n << "\n";
+    if (conjs.find(n) == conjs.end())
+    {
+      Trace("check-synth-sol") << "It is an auxiliary assertion\n";
+      auxAssertions.push_back(n);
+    }
+    else
+    {
+      Trace("check-synth-sol") << "It is a synthesis conjecture\n";
+    }
+  }
+  // check all conjectures
+  for (Node conj : conjs)
+  {
+    // Start new SMT engine to check solutions
+    std::unique_ptr<SmtEngine> solChecker;
+    initializeSubsolver(solChecker);
+    solChecker->getOptions().set(options::checkSynthSol, false);
+    solChecker->getOptions().set(options::sygusRecFun, false);
+    // get the solution for this conjecture
+    std::vector<Node>& fvars = fvarMap[conj];
+    std::vector<Node>& fsols = fsolMap[conj];
+    // Apply solution map to conjecture body
+    Node conjBody;
+    /* Whether property is quantifier free */
+    if (conj[1].getKind() != EXISTS)
+    {
+      conjBody = conj[1].substitute(
+          fvars.begin(), fvars.end(), fsols.begin(), fsols.end());
+    }
+    else
+    {
+      conjBody = conj[1][1].substitute(
+          fvars.begin(), fvars.end(), fsols.begin(), fsols.end());
+
+      /* Skolemize property */
+      std::vector<Node> vars, skos;
+      for (unsigned j = 0, size = conj[1][0].getNumChildren(); j < size; ++j)
+      {
+        vars.push_back(conj[1][0][j]);
+        std::stringstream ss;
+        ss << "sk_" << j;
+        skos.push_back(nm->mkSkolem(ss.str(), conj[1][0][j].getType()));
+        Trace("check-synth-sol") << "\tSkolemizing " << conj[1][0][j] << " to "
+                                 << skos.back() << "\n";
+      }
+      conjBody = conjBody.substitute(
+          vars.begin(), vars.end(), skos.begin(), skos.end());
+    }
+    Notice() << "SygusSolver::checkSynthSolution(): -- body substitutes to "
+             << conjBody << std::endl;
+    Trace("check-synth-sol")
+        << "Substituted body of assertion to " << conjBody << "\n";
+    solChecker->assertFormula(conjBody);
+    // Assert all auxiliary assertions. This may include recursive function
+    // definitions that were added as assertions to the sygus problem.
+    for (Node a : auxAssertions)
+    {
+      solChecker->assertFormula(a);
+    }
+    Result r = solChecker->checkSat();
+    Notice() << "SygusSolver::checkSynthSolution(): result is " << r
+             << std::endl;
+    Trace("check-synth-sol") << "Satsifiability check: " << r << "\n";
+    if (r.asSatisfiabilityResult().isUnknown())
+    {
+      InternalError() << "SygusSolver::checkSynthSolution(): could not check "
+                         "solution, result "
+                         "unknown.";
+    }
+    else if (r.asSatisfiabilityResult().isSat())
+    {
+      InternalError()
+          << "SygusSolver::checkSynthSolution(): produced solution leads to "
+             "satisfiable negated conjecture.";
+    }
+  }
+}
```

Comment:
It seems you forgot to remove the corresponding code from `smt_engine.cpp`...

Reply:
Ah right, good catch.

Reply:
`getSynthSolutions` still exists in `smt_engine.cpp`...